parquet/src/main/java/org/apache/iceberg/parquet/ApplyNameMapping.java

```diff
@@ -88,6 +96,31 @@ public Type primitive(PrimitiveType primitive) {
     return field == null ? primitive : primitive.withId(field.id());
   }
 
+  @Override
+  public void beforeElementField(Type element) {
+    super.beforeElementField(makeElement(element));
+  }
+
+  @Override
+  public void afterElementField(Type element) {
+    super.afterElementField(makeElement(element));
+  }
+
+  private Type makeElement(Type element) {
+    // List's element in 3-level lists can be named differently across different parquet writers.
+    // For example, hive names it "array_element", whereas new parquet writers names it as "element".
+    if (element.getName().equals("element") || element.isPrimitive()) {
+      return element;
+    }
```

Comment:
I think we can make it generic from the perspective that we don't have magic keywords like "element", however I am not sure if the projection change you are referring to would work here, but do let me know if you think otherwise. Here, we are making this change so that constructed parquet schema had correct ids, but still has the same schema as the one in parquet file. I will at least make this part generic, will update the diff shortly.

Reply:
I tried a bit to remove the usage of "element", however there isn't a clean way to do so. The only way I could come up with is build a dummy list and then get the element out of it. I am not sure if that is any cleaner than the current approach. Below is the change I am referring to.
```
  private Type makeElement(Type element) {
    // List's element in 3-level lists can be named differently across different parquet writers.
    // For example, hive names it "array_element", whereas new parquet writers names it as "element".
    if (element.isPrimitive()) {
      return element;
    }

    Types.BaseListBuilder.GroupElementBuilder<GroupType, Types.ListBuilder<GroupType>> dummyBuilder = Types.list(Type.Repetition.OPTIONAL)
            .groupElement(element.getRepetition())
            .addFields(element.asGroupType().getFields().toArray(new Type[0]));
    if (element.getId() != null) {
      dummyBuilder.id(element.getId().intValue());
    }
    return dummyBuilder.named("dummy").getType(0).asGroupType().getType(0);
  }
```

As such, I don't have a strong preference on it. @kbendick let me know if you like this one better and I will make the change.

Reply:
@kbendick I also updated test to dynamically generate parquet file in required format. Mind giving it another pass.

Reply:
Yeah I'll absolutely give this another pass. Sorry I was head down in something else today.

I'll also look for somebody to start the tests for you unless you'd like to wait. 😄

Reply:
And yeah, understood the projection related changes, while similar in concept, don't necessarily apply here unfortunately.

Reply:
Thanks @kbendick , getting the tests started will definitely be helpful.

Reply:
The updated tests look great @SinghAsDev! Thank you!